src/monitor.tf

```diff
@@ -62,3 +62,32 @@ resource "azurerm_monitor_action_group" "slack" {
 
   tags = var.tags
 }
+
+resource "azurerm_monitor_diagnostic_setting" "app_gw" {
+  count = var.env_short == "p" ? 1 : 0
+  # count = var.sec_log_analytics_workspace_id ? 1 : 0
+  provider           = azurerm.Prod-Sec
+  name               = format("%s-app-gw-logs", local.project)
+  target_resource_id = module.app_gw.id
+  log_analytics_workspace_id = # insert the key vault value for the log analytics workspace id 
```

Comment:
you can read the secret with this data source: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/key_vault_secret